core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java

```diff
@@ -52,6 +53,10 @@ class MethodSecurityEvaluationContext extends MethodBasedEvaluationContext {
 		super(mi.getThis(), getSpecificMethod(mi), mi.getArguments(), parameterNameDiscoverer);
 	}
 
+	MethodSecurityEvaluationContext(MethodInvocation mi, ParameterNameDiscoverer parameterNameDiscoverer) {
```

Comment:
Thanks for this cleanup. Will you please place this constructor into a separate commit since it's not directly about `Supplier<Authentication>`?

Reply:
This constructor is used in `DefaultMethodSecurityExpressionHandler.createEvaluationContext` so let’s keep it in one commit.